etc/patches/generate-android-system-wrappers.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/mozjs/config/system-headers.mozbuild b/mozjs/config/system-headers.mozbuild
+index 0872969..f3b7362 100644
+--- a/mozjs/config/system-headers.mozbuild
++++ b/mozjs/config/system-headers.mozbuild
+@@ -1048,7 +1048,7 @@ system_headers = [
+     'zmouse.h',
+ ]
+ 
+-if CONFIG['MOZ_WIDGET_TOOLKIT'] == 'android':
++if CONFIG['OS_TARGET'] == 'Android':
```

Comment:
This patch feels like it should be unnecessary. I'm asking some folks in #build about how we can investigate the root cause.

Reply:
```
12:14 <froydnj> jdm: I told ajeffrey that was the "right way" to fix things
12:15 <froydnj> jdm: but I also recall saying that patch should be submitted upstream, as it were
12:18 <froydnj> jdm: but to answer the original question, MOZ_WIDGET_TOOLKIT only gets defined in a gecko build, because toolkit/moz.configure isn't evaluated in a spidermonkey build
```

Reply:
Yes, I'm planning on submitting this upstream.